hilti/runtime/include/util.h

```diff
@@ -362,18 +364,21 @@ std::string escapeBytes(std::string_view s, bitmask<render_style::Bytes> style =
 std::string escapeUTF8(std::string_view s, bitmask<render_style::UTF8> style = render_style::UTF8::Default);
 
 /**
- * Joins elements of a container into a string, using a specified delimiter
+ * Joins elements of a range into a string, using a specified delimiter
  * to separate them.
  */
-template<typename T>
-std::string join(const T& l, const std::string& delim = "") {
+template<std::ranges::input_range T>
+std::string join(T&& l, std::string_view delim = "")
+    requires(std::is_constructible_v<std::string, std::ranges::range_value_t<T>>)
+{
     std::string result;
     bool first = true;
 
     for ( const auto& i : l ) {
         if ( not first )
```

Comment:
while here could we change this to `! first`? It's one of those cases that compilers differ in behavior and I find it unclear. I know this wasn't touched so doesn't have to be here, just a request